### ISSUE TITLE
Continuing training

### ISSUE THREAD

**Issue by amfogor** [ultralytics/yolov5#3703]
If I train on a custom dataset, what is the correct way to continue training for a given checkpoint?

So if I trained with the following command
```
python yolov5/train.py \
    --epochs 1000 \
    --cfg yolov5/models/yolov5s.yaml \
    --img-size 1280 1280 \
    --data train/dataset.yaml \
    --hyp hyp.my.yaml
```

and trained until finished, then used the following command to train some more

```
python yolov5/train.py \
    --epochs 1000 \
    --cfg yolov5/models/yolov5s.yaml \
    --img-size 1280 1280 \
    --data train/dataset.yaml \
    --hyp hyp.my.yaml \
    --weights runs/train/exp/weights/best.pt
```

loss values would start higher than expected.
I expect them to be the same as they were on the last epoch of training.
Here are the values of the last epoch
```
    box       obj       cls     total
0.01376    0.0918  0.005228    0.1108 
```
And here are the values  of the first epoch when continuing training
```
    box       obj       cls     total
0.07962    0.3484  0.007278    0.4353
```

So I suspect that hyperparameters are evolved during training. Is that so?

What am I doing wrong or not understanding here?

**Comment by glenn-jocher** [comment]
@amfogor 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck and let us know if you have any other questions!

**Comment by amfogor** [comment]
@glenn-jocher but if I don't specify the `--hyp` option, `yolov5/data/hyp.finetune.yaml` will be used. But these parameters are not for a custom dataset.

**Comment by amfogor** [comment]
@glenn-jocher oh, I see, on the first epochs learning rate is high, and that's the reason why losses begin to deviate so much.

**Comment by amfogor** [comment]
@glenn-jocher what hyperparameters should I change and by how much when continuing training?
